test/step-function.localstack.test.ts

```diff
@@ -1105,4 +1167,37 @@ localstackTestSuite("sfnStack", (testResource, _stack, _app) => {
     },
     "error1Error2error3finally1setfinally2error4watforwatfinallyrecatcherror6whilewatdowatsfnmapwatarrmapwat"
   );
+
+  test(
+    "for control and assignment",
+    (parent) => {
+      return new StepFunction<{ arr: number[] }, string>(
+        parent,
+        "fn",
+        async (input) => {
+          let a = "";
+          for (const i in input.arr) {
+            if (i === "2") {
```

Comment:
comparing with string?

Reply:
`i` is the index which is a string. see the for of below comparing with a number.